framework/data/BaseDataProvider.php

```diff
@@ -44,6 +50,20 @@
     private $_keys;
     private $_models;
     private $_totalCount;
+    
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        parent::init();
+        if (!isset($this->id)) {
+            if (self::$counter > 0) {
```

Comment:
Okay, but then this issue will remain unsolved until the whole framework gets changed to use this new trait meaning it will touch everything and is a much bigger operation.